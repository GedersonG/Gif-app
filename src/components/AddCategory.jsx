import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ( {target } ) => {
        setInputValue( target.value );
    }

    const onSubmit = (e) => {
        e.preventDefault();
        const value = inputValue.trim();
        if( value.length <= 1 ) return; // op.

        onNewCategory( value );
        setInputValue('');
    }

    return (
        <form onSubmit={ (e) => onSubmit(e) }>
            <input
            type="text"
            placeholder="Search gifs"
            value={ inputValue }
            onChange={ onInputChange }
            />
        </form>
    );
}
