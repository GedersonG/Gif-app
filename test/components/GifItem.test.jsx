import { render, screen, fireEvent } from '@testing-library/react';

import { GifItem } from "../../src/components/GifItem";

describe('Test to GifItem component', () => { 
    
    const title = 'This is a title';
    const url = 'https://myurl.com/';

    test('should take a snapshot to GifItem component', () => { 
        
        const { container } = render(
            <GifItem 
                title = { title }
                url = { url } 
            /> 
        );
        expect( container ).toMatchSnapshot();

     });

     test('should show the gif with its url', () => { 

        render(
            <GifItem 
                title = { title }
                url = { url } 
            /> 
        );

        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( title );
     });

     test('should show the title in the component', () => { 

        render(
            <GifItem 
                title = { title }
                url = { url } 
            /> 
        );

        expect( screen.getByText( title )).toBeTruthy();

     });

 })