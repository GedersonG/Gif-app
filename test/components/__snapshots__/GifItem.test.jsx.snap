// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to GifItem component should take a snapshot to GifItem component 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="This is a title"
      src="https://myurl.com/"
    />
    <p>
      This is a title
    </p>
  </div>
</div>
`;
